File: generators/java8/src/main/java/io/symphonia/DataGenerator.java
package io.symphonia;

import com.amazonaws.services.lambda.runtime.Context;

import java.util.Date;

@SuppressWarnings("unused")
public class DataGenerator {
    public String handler(Object o, Context context) throws InterruptedException {
        String message = String.format("Lambda with %dMB memory invoked at %tc", context.getMemoryLimitInMB(), new Date());
        Thread.sleep(50);
        System.out.println(message);
        return message;
    }
}

File: generators/node8/src/dataGenerator.js
exports.handler = async (event, context) => {
    const message = `Lambda with ${context.memoryLimitInMB}MB memory invoked at ${new Date()}`
    await sleep(50);
    console.log(message);
    return message;
};

function sleep(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}